graphql/resolver.go

```diff
@@ -35,6 +36,46 @@ func New(apiURL string) Config {
 			sc: dbConnector,
 		},
 	}
+	c.Directives.RequireHostAccess = func(ctx context.Context, obj interface{}, next graphql.Resolver, access HostAccessLevel) (interface{}, error) {
+		args, isStringMap := obj.(map[string]interface{})
+		if !isStringMap {
+			return nil, ResourceNotFound.Send(ctx, "host not specified")
+		}
+		hostId, hasHostId := args["hostId"].(string)
+		hostIdsInterface, hasHostIds := args["hostIds"].([]interface{})
+		hostIds := []string{}
+		if hasHostIds {
+			for _, v := range hostIdsInterface {
+				hostIds = append(hostIds, v.(string))
+			}
+		}
+		if !hasHostId && !hasHostIds {
+			return nil, ResourceNotFound.Send(ctx, "host not specified")
+		}
```

Comment:
I think it makes sense to switch the order, since we might not end up using `hostIds` if we hit this if statement
```suggestion
		if !hasHostId && !hasHostIds {
			return nil, ResourceNotFound.Send(ctx, "host not specified")
		}
		
		hostIds := []string{}
		if hasHostIds {
			for _, v := range hostIdsInterface {
				hostIds = append(hostIds, v.(string))
			}
		}
```